packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js

```diff
@@ -227,8 +227,7 @@ const CustomerForm = props => {
                     <Region
                         validate={isRequired}
                         fieldInput={'region[region]'}
-                        fieldSelect={'region[region_id]'}
-                        optionValueKey="id"
+                        fieldSelect={'region[code]'}
```

Comment:
@tjwiebell I know this isn't the solution you suggested and your other thoughts but I believe those are becoming complex given it involves changing a fragment that is used by multiple components. So I resorted to this solution which is the simple one for time being till graphql team makes the change and at that point, we will have to change the default value of `optionValueKey` in the `region talon` and everything will start working fine.

Hope this explanation makes a valid point to stick with this solution.

Reply:
The CustomerAddress field is `region[region_code]`, so this fix ends up breaking a lot of other functionality (can't update or create new addresses in the Checkout AddressBook). I think even switching that here would bring back the bug this intends to solve, since the data from the Cart wouldn't line up. This might just require that breaking change to the fragment.

![Screen Shot 2021-04-08 at 9 15 21 AM](https://user-images.githubusercontent.com/462953/114044080-81599400-984c-11eb-9cc9-8a05a103060d.png)